jetty-http/src/main/java/org/eclipse/jetty/http/UriCompliance.java

```diff
@@ -84,19 +92,29 @@ public String getDescription()
     }
 
     /**
-     * The default compliance mode that extends RFC3986 compliance with additional violations to avoid ambiguous URIs
+     * The default compliance mode that extends RFC3986 compliance with additional violations to avoid most ambiguous URIs.
+     * This mode does allow {@link Violation#AMBIGUOUS_PATH_SEPARATOR}, but disallows
+     * {@link Violation#AMBIGUOUS_PATH_PARAMETER} and {@link Violation#AMBIGUOUS_PATH_SEGMENT}.
+     * Ambiguous paths are not allowed by {@link Violation#NON_CANONICAL_AMBIGUOUS_PATHS}.
      */
-    public static final UriCompliance DEFAULT = new UriCompliance("DEFAULT", noneOf(Violation.class));
+    public static final UriCompliance DEFAULT = new UriCompliance("DEFAULT", of(Violation.AMBIGUOUS_PATH_SEPARATOR));
 
     /**
      * LEGACY compliance mode that disallows only ambiguous path parameters as per Jetty-9.4
      */
-    public static final UriCompliance LEGACY = new UriCompliance("LEGACY", EnumSet.of(Violation.AMBIGUOUS_PATH_SEGMENT, Violation.AMBIGUOUS_PATH_SEPARATOR));
+    public static final UriCompliance LEGACY = new UriCompliance("LEGACY", of(Violation.AMBIGUOUS_PATH_SEGMENT, Violation.AMBIGUOUS_PATH_SEPARATOR));
 
     /**
-     * Compliance mode that exactly follows RFC3986, including allowing all additional ambiguous URI Violations
+     * Compliance mode that exactly follows RFC3986, including allowing all additional ambiguous URI Violations. However ambiguous paths are
+     * canonicalized for safety.
      */
-    public static final UriCompliance RFC3986 = new UriCompliance("RFC3986", allOf(Violation.class));
+    public static final UriCompliance RFC3986 = new UriCompliance("RFC3986", complementOf(of(Violation.NON_CANONICAL_AMBIGUOUS_PATHS)));
+
+    /**
+     * Compliance mode that exactly follows RFC3986, including allowing all additional ambiguous URI Violations, which may be passed
+     * to the application in non-normal form.
```

Comment:
And this description too should probably be clear about which `Violation` types we allow here.